common/sdk/headers_provider.go

```diff
@@ -0,0 +1,25 @@
+package sdk
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type HeadersProvider interface {
```

Comment:
This interface is internal in the go-sdk for some reason :(
So, I just copied it for now